api/v1alpha1/bundleinstance_types.go

```diff
@@ -46,18 +46,28 @@ const (
 type BundleInstanceSpec struct {
 	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
 	ProvisionerClassName string `json:"provisionerClassName"`
+	// Label selector for Bundles resources.
+	Selector *metav1.LabelSelector `json:"selector"`
+	// Template describes the generated Bundle that this instance will manage.
+	Template *BundleTemplate `json:"template"`
+}
 
-	// BundleName is the name of the bundle that this instance is managing on the cluster.
-	BundleName string `json:"bundleName"`
+// BundleTemplate defines the desired state of a Bundle resource
+type BundleTemplate struct {
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the desired behavior of the Bundle.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
```

Comment:
Is there a use-case where the template is provided but the spec is omitted? ie should the spec be optional?

Reply:
Seems like the adoption use case maybe?

Reply:
I had copied the [PodTemplateSpec](https://github.com/kubernetes/api/blob/a2ee8c7e6bdc1744015c35711607d5403bbd1d39/core/v1/types.go#L3953) type definitions for this, but it's unclear to me what the rationale was making the spec field optional (and following the git blame trail doesn't help illuminate that decision either).

I'm fine with making this required until we have a concrete use case?

Reply:
Actually: does leaving it optional give us more leeway here? Wouldn't making it required -> optional down the line to support something like adoption require us to make a breaking change to the API?

Reply:
required -> optional is non-breaking because all existing instance will have set it already.
optional -> required is breaking because there will be some instances that have not set it, and now we're requiring them to be set.

Reply:
Updated to make spec required in the last force push.